pydantic_ai_slim/pydantic_ai/models/google.py

```diff
@@ -511,7 +527,18 @@ def _process_response_from_parts(
 ) -> ModelResponse:
     items: list[ModelResponsePart] = []
     for part in parts:
-        if part.text is not None:
+        if part.executable_code is not None:
+            items.append(ServerToolCallPart(args=part.executable_code.model_dump(), tool_name='code_execution'))
```

Comment:
We should set `model_name='google'` here (or `provider_name`, if you rename it as I suggested above)